start-from-zero/Javascript/src/views/index/index.js

```diff
@@ -5,19 +5,68 @@ function redirectToEvaluation(){
         e.stopPropagation();
         e.preventDefault();
         app.innerHTML = EvaluationsPage();
-        redirectToLogin();
+        
         redirectToNewEvaluation();
     });
 }
 
 function redirectToNewEvaluation(){
+    app.innerHTML = EvaluationsPage();
     const newEvaluations = document.querySelector('.new-evaluations-page');
     newEvaluations.addEventListener('click',function(e){
         e.stopPropagation();
         e.preventDefault();
         app.innerHTML = NewEvaluationPageAll();
-        redirectToLogin();
+        
         redirectToEvaluation();
+
+        const btn = document.getElementById('btn');
+        var form = document.getElementById('form-new');
+
+        btn.addEventListener('click',function(){
+            function User(username,password,date,checkbox,textarea,select){
+            this.username = username;
+            this.password = password;
+            this.date = date;
+            this.checkbox = checkbox
+            this.textarea= textarea;
+            this.select = select;
+            }
+
+            const username = document.getElementById('username-id').value;
+            const password = document.getElementById('password-id').value;
+            const calendar = document.getElementById('calendar-id').value;
+            const checkbox = document.querySelectorAll('.input-checkbox');
+            const textarea = document.querySelectorAll('textarea');
+            const select = document.querySelectorAll('select');
+            const option = document.querySelectorAll('option');
+
+            const textareaArr = [];
+            textarea.forEach(element => {
```

Comment:
This approach gives you a representation of all the data in the form, but I don't think it is a complete one. I would store this information in an object with keys representing each textarea / input / select etc so that the information is not so stringly coupled with the element order index. This way, if the form structure changes (at any point) you don't need to make changes to code in other areas besides the views.

Reply:
Thanks for the advice Sebi. :)